File: 388-longest-absolute-file-path/leochang/388.cpp
class Solution {
public:
	int lengthLongestPath(string input) {
		int maxLen = 0;
		map <int, int> len;

		for(int i = 0; i < input.size(); i++){
			int count = 0, dirLevel = 1;
			bool isFile = false;
			while(input[i] == '\t' && i < input.size()){
				dirLevel++;
				i++;
			}
			while(input[i] != '\n' && i < input.size()){
				if (input[i] == '.') isFile = true;
				count++;
				i++;
			}
			if (isFile) {
				maxLen = max(maxLen, len[dirLevel - 1] + count);
			} else {
				len[dirLevel] = len[dirLevel - 1] + count + 1;
			}
		}
		return maxLen;
	}
};

File: 388-longest-absolute-file-path/ken/388.py
class Solution(object):
    def lengthLongestPath(self, input):
        """
        :type input: str
        :rtype: int
        """

        prefix = []
        inputs = input.split("\n")
        longest = 0

        for i in inputs:
            tokens = i.split("\t")

            while len(prefix) >= len(tokens):
                prefix.pop()
            prefix.append(tokens[-1])

            if "." in tokens[-1]:
                path = "/".join(prefix)
                longest = max(longest, len(path))

        return longest


a = Solution()
a.lengthLongestPath("dir\n\tsubdir1\n\tsubdir2\n\t\tfile.ext")
